src/app/lib/views/torrent_collection.js

```diff
@@ -76,179 +107,193 @@
                 return;
             }
 
+            if($('.search_in').is(':visible')) {
+                $('.togglesengines').removeClass('fa-caret-up').addClass('fa-caret-down');
+                $('.search_in').css('display', 'none');
+            }
+
             $('.onlinesearch-info>ul.file-list').html('');
 
             $('.online-search').removeClass('fa-search').addClass('fa-spin fa-spinner');
+            $('.togglesengines').css('visibility', 'hidden');
 
             var index = 0;
             console.warn(category);
 
-            var leetx = function () {
-                return new Promise(function (resolve) {
-                    var results = [];
-                    setTimeout(function () {
-                        resolve(results);
-                    }, 6000);
-                    var leet = torrentCollection.leet;
-                    leet.search({
-                        query: input.toLocaleLowerCase(),
-                        category: category,
-                        orderBy: 'seeders',
-                        sortBy: 'desc',
-                    }).then(function (data) {
-                        console.debug('1337x search: %s results', data.torrents.length);
-                        var indx = 1, totl = data.length;
-                        data.torrents.forEach(function (item) {
-                            leet.info('https://1337x.to' + item.href).then(function (ldata) {
+            var piratebay = function () {
+                if (Settings.enablethepiratebay) {
+                    return new Promise(function (resolve) {
+                        var results = [];
+                        setTimeout(function () {
+                            resolve(results);
```

Comment:
why is this here? is this supposed to be a 6 second timeout on searches? if so, please add a comment. if not, you should be calling resolve once you have all the results. we're in node v12, so we can use async functions as well

Reply:
That wasnt me. It was pre-existing code.

Reply:
Oh, I wasn't aware of that. Where is this code from then?